tests/Test.ADAL.NET.Integration/FederatedUserCredentialTests.cs

```diff
@@ -320,7 +333,18 @@ public void Initialize()
             Assert.AreEqual(1, context.TokenCache.Count);
 
             // Cache entry updated with new access token
-            Assert.AreEqual("some-access-token", context.TokenCache.tokenCacheDictionary[key].Result.AccessToken);
+            var entry = await context.TokenCache.LoadFromCache(new CacheQueryData
```

Comment:
instead of asking cache to load, you should explicitly examine the dictionary. that would circumvent any false positives that LoadFromCache() may throw